api/api.proto

```diff
@@ -645,27 +645,87 @@ message StorageCluster {
 }
 
 service OpenStorageCluster {
-	// Enumerate lists all the nodes in the cluster.
-	rpc Enumerate(ClusterEnumerateRequest)
-		returns (ClusterEnumerateResponse) {}
+  // Enumerate lists all the nodes in the cluster.
+  rpc Enumerate(ClusterEnumerateRequest)
+    returns (ClusterEnumerateResponse) {}
 
-	// Inspect the node given a UUID.
-	rpc Inspect(ClusterInspectRequest)
+  // Inspect the node given a UUID.
+  rpc Inspect(ClusterInspectRequest)
     returns (ClusterInspectResponse) {}
-    
+
   // Get a list of alerts from the storage cluster
   rpc AlertEnumerate(ClusterAlertEnumerateRequest)
     returns (ClusterAlertEnumerateResponse) {}
-    
+
   // Clear the alert for a given resource
   rpc AlertClear(ClusterAlertClearRequest)
     returns (ClusterAlertClearResponse) {}
-  
+
   // Erases an alert for a given resource
   rpc AlertErase(ClusterAlertEraseRequest)
     returns (ClusterAlertEraseResponse) {}
 }
 
+service OpenStorageVolume {
+  // Creates a volume
+  rpc Create(VolumeCreateRequest)
+    returns (VolumeCreateResponse) {}
+
+  // CreateSimpleVolume provides a simple API to create a volume
+  rpc CreateSimpleVolume(VolumeCreateSimpleVolumeRequest)
```

Comment:
IMO Providing two Create APIs would unnecessarily create confusion for the developers. Also exposing this CreateSimpleVolume API can be error prone as the developers would have to pass all the volume attributes as name value pairs and they can make mistakes. The existing Create API is well defined with an Object that can be passed over grpc/http and hence less error prone.

For docker/kubernetes it is OSD who is making the conversion from map to an object and hence it is under our control and less error prone.

Reply:
Here is a more concrete proposal of the proto message:

Reply:
I think we should use the same API as we have currently (and use the same VolumeSpec and other proto definitions).

Reply:
I really would prefer for the users of this API not to use the three objects. It is extremely confusing and some of the values in api.VolumeSpec are not even used. I would like to provide a consistent clear API for them to consume. This API does not hinder the current one we use on REST at all, but when we expose these APIs to users which are new to OpenStorage, they would prefer a simpler, clearer API call. I know, because, everything I see the Create call, it takes me a while to understand it, and is only by looking into the source code as examples. I do not want to put this burden on the developers.